src/core/cgroup.c

```diff
@@ -4075,25 +4075,33 @@ int unit_get_memory_peak(Unit *u, uint64_t *ret) {
         if (!UNIT_CGROUP_BOOL(u, memory_accounting))
                 return -ENODATA;
 
-        r = unit_get_memory_peak_raw(u, &bytes);
+        r = unit_get_memory_raw_generic(u, &bytes, mem_attribute);
         if (r == -ENODATA && u->memory_peak_last != UINT64_MAX) {
                 /* If we can't get the memory peak anymore (because the cgroup was already removed, for example),
                  * use our cached value. */
 
                 if (ret)
-                        *ret = u->memory_peak_last;
+                        *ret = *last;
```

Comment:
So I think `last` is not actually needed. Instead, just update the cache through this function, and return the corresponding cached value in unit_get_memory_{,swap_}peak.

Reply:
> So I think `last` is not actually needed.

It isn't, but your suggested alternative increases the size of two functions from one code line to seven. I also do not see the benefit of doing so. Therefore, I personally, would go with the current approach.

However, if you feel like this change is required to get it merged, then I will happily comply.